frontend/src/services/profile_handler_service.ts

```diff
@@ -0,0 +1,54 @@
+import { PersonProps, MockProfileBackendService } from "./mock_profile_backend";
+
+/** Error that occurs when Id does not exist */
+export class NonExistentProfileError extends Error {
+  constructor(readonly id: string) {
+    super(`Profile ${id} does not exist`);
+  }
+}
+
+/** Error that occurs if Id was unable to be added */
+export class FailureToUpdateProfile extends Error {
+  constructor(readonly id: string) {
+    super(`Unable to update Profile ${id}`);
+  }
+}
+
+interface ProfileHandlerServiceAPI {
+  getPerson(id: string): Promise<PersonProps>;
+  updatePerson(PersonProps): Promise<boolean>;
+  deletePerson(id: string): Promise<boolean>;
+}
+
+/**
+ * Profile Handling service that manages list of user profiles using a mocked
+ * backend database.
+ */
+export class ProfileHandlerService implements ProfileHandlerServiceAPI {
```

Comment:
nit: The usual terminology for this is `ProfileHandlerServiceInterface`. API is kind of an overloaded term, in web development it usually refers to the server API, not a local code interface

Reply:
Noted